### ISSUE TITLE
Signup fails: null value in column "provider_id" of relation "identities" violates not-null constraint

### ISSUE THREAD

**Issue by jaulz** [supabase/supabase#19620]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

As of today, whenever I try to sign up I see this error in my local development environment:
````
null value in column "provider_id" of relation "identities" violates not-null constraint
````

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Run `supabase db reset`
2. Register via `signUp` function
```typescript      
const { data, error } = await supabase.auth.signUp({
        email,
        password,
      })
```
4. Check error return value which contains above mentioned value

## Expected behavior

Registration with email and password should work without issues

## Screenshots

n/a

## System information

- OS: macOS
- Browser (if applies): chrome
- Version of supabase-js: 2.39.0
- Version of Node.js: 21.0.0

## Additional context

My current workaround is to remove the null constraint:
```sql
alter table auth.identities alter column provider_id drop not null;
```

**Comment by JunsikChoi** [comment]
I have same issue.
Production `auth.identities` table does not match with local supabase `auth.identities` table.
I can not use the seed.sql for creating test users.

**Comment by J0** [comment]
Hey folks,

Thanks for flagging this! Could we trouble y'all to open a ticket [here](https://supabase.com/dashboard/support/new) If you could state any relevant details as well as the framework(s)/tools you are using that would be great (e.g. nextjs with prisma, django with sqlalchemy, redwoodjs)

Thanks!

**Comment by hf** [comment]
@JunsikChoi Hmm... that's good information. We actually just finished releasing a version to all projects on prod, but CLI moves slightly faster than the prod deploys.

Can you check again? 

I'll speak to the CLI team on making the releases synced.

**Comment by sweatybridge** [comment]
I suspect this error is specific to projects on pg14. Please give `npx supabase@beta start` a try.

Feel free to reopen if this issue still persists.

**Comment by paynwahs** [comment]
Hi, the same issue persists on pg15.
Production auth.identities table does not match with local supabase auth.identities table (Both running pg15).
I cannot use the seed.sql for creating test users.

Problem persist even after running `Supabase start` using the latest [beta release v1.123.6](https://github.com/supabase/cli/releases/tag/v1.123.6)

**Comment by Shusanta** [comment]
yea my build processes are all failing with the exact same errors /:

**Comment by AndreMiras** [comment]
I got the issue when running `npx supabase db reset` after updating the CLI from 1.110.1 to 1.123.4 and my DB contains a seed with the following insert:
```sql
INSERT INTO auth.identities (id, user_id, identity_data, provider, last_sign_in_at, created_at, updated_at) VALUES (...);
```
Notice how we provide the provider (which used to be a string) and the schema most likely changed to provider_id being a fk.
I gave `npx supabase@beta start` a try but it ends up with an error:
```
...
Status: Downloaded newer image for public.ecr.aws/supabase/postgres:15.1.0.147
database is not healthy
Try rerunning the command with --debug to troubleshoot the error.
```
Not big deal for local dev, I'll simply downgrade to a CLI version that works

**Comment by kangmingtay** [comment]
@AndreMiras interesting, the auth team introduced a [migration](https://github.com/supabase/gotrue/compare/v2.115.1...v2.116.0) to the auth service which makes the following changes:
* Remove the composite primary key on (`auth.identities.id`, `auth.identities.provider`)
* Rename `auth.identities.id` to `auth.identities.provider_id` 
* Create a unique constraint on (`auth.identities.provider_id`, `auth.identities.provider`)
* Add a new primary key called `auth.identities.id`

This seems to affect `seed.sql` files that were relying on the old schema of the `auth.identities` table. I think it should work if you update all lines that contain:

```sql
INSERT INTO auth.identities (id, user_id, identity_data, provider, last_sign_in_at, created_at, updated_at)
```
to 
```sql
INSERT INTO auth.identities (provider_id, user_id, identity_data, provider, last_sign_in_at, created_at, updated_at)
```

**Comment by AndreMiras** [comment]
That did it @kangmingtay, thanks for the context

**Comment by jaulz** [comment]
@kangmingtay i don't have anything in seed.sql that inserts into identities table. Are there any other spots as well? 🤔

**Comment by EAnkerstjerne** [comment]
> I suspect this error is specific to projects on pg14. Please give `npx supabase@beta start` a try.
> 
> Feel free to reopen if this issue still persists.

I have the issue that my local setup auth tables are not up to date with the new provider_id column in identities table. I have "major_version = 15" in supabase/config.toml. I have tried to run it with beta version of CLI and latest release version, no changes. BUT when I change major version to 14 "major_version = 14", it works as expected with the new provider_id column. It would seem the initial schemas on pg15 are not updated on the CLI.

Hope this helps in some way. ◡̈

**Comment by paynwahs** [comment]
Are there plans to upgrade the initial schemas on pg15 for Supabase CLI soon?

**Comment by jagatfx** [comment]
It appears https://github.com/supabase/gotrue/pull/1311 should have also had associated `supabase/gotrue` docker images. It may be good to include the docker image flow in the automated tests.

**Comment by sweatybridge** [comment]
> Are there plans to upgrade the initial schemas on pg15 for Supabase CLI soon?

pg15 doesn't require initial schema because we run the migrations that came with gotrue docker image.

It's possible that you have have outdated image versions in `supabase/.temp` directory locally. Try deleting the directory to reset to the defaults, ie. `rm -rf supabase/.temp`.

**Comment by EAnkerstjerne** [comment]
> > Are there plans to upgrade the initial schemas on pg15 for Supabase CLI soon?
> 
> pg15 doesn't require initial schema because we run the migrations that came with gotrue docker image.
> 
> It's possible that you have have outdated image versions in `supabase/.temp` directory locally. Try deleting the directory to reset to the defaults, ie. `rm -rf supabase/.temp`.

Thanks. This solved it for me!

I was thinking all along that I had outdated images. Just didn't know it was the .temp folder that held that information. Tried deleting all images and containers without luck. But this did the trick!

**Comment by robertsamarji** [comment]
Is there a solution to this other than downgrading the CLI version?

Using Supabase CLI v1.123.4 with pg15, removed ```supabase/.temp```, reset the database and still getting the error (below) when signing up new users with ```supabase.auth.signInWithOtp```.

```"null value in column \"provider_id\" of relation \"identities\" violates not-null constraint"```

**Comment by svapreddy** [comment]
Same here, I don't have anything under `supabase/.temp` and I don't have anything in `seed.sql` that touches `auth.identities`. I am using the latest CLI version `^1.123.4`.

Still signup fails.

**Comment by svapreddy** [comment]
I noodled a lil bit, and I got to this error now:

```
ERROR: column "provider_id" of relation "identities" already exists (SQLSTATE 42701)  
At statement 2: alter table "auth"."identities" add column "provider_id" text not null
```

**Comment by paynwahs** [comment]
Using Supabase CLI v1.123.6 with pg15, removed supabase/.temp, reset the database and still getting the error (below) when signing up new users on local with [supabase.auth.signUp](https://supabase.com/docs/guides/auth/auth-email).

`AuthApiError: null value in column "provider_id" of relation "identities" violates not-null constraint`.

I am able to use `seed.sql` to populate local Supabase instance using the following code, thus verifing the latest pg15 schema.

`INSERT INTO auth.identities (provider_id, user_id, identity_data, provider, last_sign_in_at, created_at, updated_at)`
 
I am using the `@Supabase/ssr` package and its `createServerClient()` function to create the Supabase client instance. I then use this instance to call `supabase.auth.signUp` when the `AuthApiError` occur. 

Could this be a bug on the `@Supabase/ssr` package as it is unable to sign up new users while conforming to the pg15 `auth.identities` schema?

**Comment by artemgordash** [comment]
@paynwahs update to [v1.128.1](https://github.com/supabase/cli/releases/tag/v1.128.1) solved for me sign up on local

**Comment by paynwahs** [comment]
> @paynwahs update to [v1.128.1](https://github.com/supabase/cli/releases/tag/v1.128.1) solved for me sign up on local

Thanks. This solved it for me!

**Comment by encima** [comment]
Closing this as it seems to be solved, confirmed working on v1.128.1 also.

Please comment here (with system information) if this is still occurring and the fix mentioned [here](https://github.com/supabase/supabase/issues/19620#issuecomment-1862304039) does not work and we will reopen!

**Comment by voyagebagage** [comment]
I have my docker always throw me the same error when I restart the database or yarn setup 

LOG:  could not open configuration file "/etc/postgresql-custom/read-replica.conf": No such file or directory
FATAL:  configuration file "/etc/postgresql/postgresql.conf" contains errors

and the terminal error with database unhealthy

**Comment by austerj** [comment]
I was running into this issue as well (despite an empty `seed.sql` file). It turned out to be related to some lingering containers and I solved it with steps similar to those in https://github.com/supabase/cli/issues/1922, though I had to run `docker system prune -f` first, after which `docker volume prune -a` finally got rid of the lingering supabase containers. After this I could run my local supabase container again.
